### PR TITLE
docs: add outline for how to make a blostream rollup

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

This PR adds an outline for how get started with a blobstream rollup by outlining the structure of a go implementation.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: docs/developers/blobstream-rollup-outline.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
+# Blobstream Demo Rollup
+
+Rollups can use the Blobstream for DA by posting their data to Celestia and then
+proving that it was posted on Ethereum. This is done identically to how any
+rollup or user would post data to Celestia, and then the validators sign over
+additional commitments that are relayed to Ethereum via a light client relay
+(aka Blobstream!). This demo will outline (note outline is not an
+implementation! please do not expect to copy and paste this code 🙂) a very
```
Comment (jcstein) [inline docs/developers/blobstream-rollup-outline.md:8]: :b:

File: docs/developers/blobstream-rollup-outline.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+Therefore we can start by first defining the reading and writing interations
+rollup nodes will have with both the Celestia and Ethereum networks. The actual
+implementations of these interfaces are left as excercises to the reader (🤪).
+Assume that those implementations of these interfaces are verifying the
+respective chain. For the connection to Celestia, this would likely mean
+connecting to a Celestia Light Node, which can detect faults in consensus such
+as hidden data. For the connection to Ethereum, this would likely mean running
+and connecting to a fullnode. More information on the RPC that is exposed by a
+Celestia Light Node can be found
+[here](https://node-rpc-docs.celestia.org/?version=v0.11.0), and more
```
Comment (SweeXordious) [inline docs/developers/blobstream-rollup-outline.md]:
```suggestion
in the [rpc documentation](https://node-rpc-docs.celestia.org/?version=v0.11.0), and more
```

File: docs/developers/blobstream-rollup-outline.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+information on running one can be found
+[here](https://docs.celestia.org/nodes/light-node/).
```
Comment (SweeXordious) [inline docs/developers/blobstream-rollup-outline.md]:
```suggestion
in the [running a light node documentation](https://docs.celestia.org/nodes/light-node/).
```

**Review by jcstein — APPROVED** [review]

**Comment by jcstein** [comment]
[Preview of page](https://docs.celestia.org/pr-preview/pr-1155/developers/blobstream-offchain/)

File: docs/developers/blobstream-rollup-outline.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+
+For optimistic rollups, this could be as simple as referencing the data in the
+celestia block, not unlike using a pointer in memory. This is what is done below
+via a `Span` in the [Creating Blocks](#Creating-Blocks) section. We keep track
+of where the data is located in the celestia block and the sequencer signs over
+that location in the header. If the sequencer commits to non-existent data or an
+invalid state root, then the invalid transaction is first proved to be included
+in the `Span` before the rest of the fraud proof process is followed. More
+information on Blobstream fraud proofs can be found
+[here](https://github.com/celestiaorg/blobstream-contracts/blob/v3.0.0/docs/inclusion-proofs.md#blobstream-fraud-proofs).
```
Comment (SweeXordious) [inline docs/developers/blobstream-rollup-outline.md]:
```suggestion
in the [inclusion proofs documentation](https://github.com/celestiaorg/blobstream-contracts/blob/v3.0.0/docs/inclusion-proofs.md#blobstream-fraud-proofs).
```

File: docs/developers/blobstream-rollup-outline.md [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+
+For zk rollups, this would involve creating an inclusion proof to the data root
+tuple root in the Blobstream contracts, and then verifying that proof in the zk
+proof used to verify state. More information on inclusion proofs can be found
+[here](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md#1-data-root-inclusion-proof).
```
Comment (SweeXordious) [inline docs/developers/blobstream-rollup-outline.md]:
```suggestion
in the [data root inclusion proof documentation](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md#1-data-root-inclusion-proof).
```

File: docs/developers/blobstream-rollup-outline.md [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+
+There are a few different mechanisms that could be used to download blocks. The
+simplest solution and what is outlined above is for Fullnodes to wait until the
+blocks and the headers are posted to the respective chains, and then download
+each as they are posted. It would also be possible to gossip the headers ahead
+of time, and download the rollup blocks from Celestia instead of waiting for the
+headers to be posted to Ethereum. It's also possible to download the headers and
+the block data like a normal blockchain via a gossiping network and only fall
+back to downloading the data and headers from Celestia and Ethereum if the
+gossiping network is unavailable or the Seuqencer is malicious.
```
Comment (SweeXordious) [inline docs/developers/blobstream-rollup-outline.md]:
```suggestion
gossiping network is unavailable or the Sequencer is malicious.
```

**Comment by SweeXordious** [comment]
@jcstein Was late for review, sorry. Where can I put these changes? just cosmetics? should I just open a new PR?

**Comment by jcstein** [comment]
i'm adding them to John's open PR now

File: docs/developers/blobstream-rollup-outline.md (hunk 2 above)
Comment (jcstein) [reply docs/developers/blobstream-rollup-outline.md]: this is done [here](https://github.com/celestiaorg/docs/blob/f8cb1a50eca74d663d3fd19a3679c7274fe7707e/docs/developers/blobstream-offchain.md?plain=1#L105)

File: docs/developers/blobstream-rollup-outline.md (hunk 3 above)
Comment (jcstein) [reply docs/developers/blobstream-rollup-outline.md]: same with this one 😄 https://github.com/celestiaorg/docs/blob/f8cb1a50eca74d663d3fd19a3679c7274fe7707e/docs/developers/blobstream-offchain.md?plain=1#L107

File: docs/developers/blobstream-rollup-outline.md (hunk 5 above)
Comment (jcstein) [reply docs/developers/blobstream-rollup-outline.md]: 98018e5170786f86d39c4609432b7e04f1e6f248

File: docs/developers/blobstream-rollup-outline.md (hunk 6 above)
Comment (jcstein) [reply docs/developers/blobstream-rollup-outline.md]: 98018e5170786f86d39c4609432b7e04f1e6f248

File: docs/developers/blobstream-rollup-outline.md (hunk 4 above)
Comment (jcstein) [reply docs/developers/blobstream-rollup-outline.md]: 98018e5170786f86d39c4609432b7e04f1e6f248

**Pull request merged**